File: lib/opencoinage/vocabulary.rb
module OpenCoinage
  ##
  # An RDFS vocabulary for digital currency issuance.
  #
  # @see http://opencoinage.org/rdf/
  # @see http://rdf.rubyforge.org/RDF/Vocabulary.html
  class Vocabulary < RDF::Vocabulary('http://opencoinage.org/rdf/')
    property :amount
    property :expires
    property :identifier
    property :issuer
    property :signature
  end # Vocabulary
end # OpenCoinage

File: src/java/org/opencoinage/Vocabulary.java
package org.opencoinage;

import java.net.URI;
import java.net.URISyntaxException;

/**
 * An RDFS vocabulary for digital currency issuance.
 *
 * @see http://opencoinage.org/rdf/
 */
public class Vocabulary {
  public static final String BASE       = "http://opencoinage.org/rdf/";
  public static final String AMOUNT     = BASE + "amount";
  public static final String EXPIRES    = BASE + "expires";
  public static final String IDENTIFIER = BASE + "identifier";
  public static final String ISSUER     = BASE + "issuer";
  public static final String SIGNATURE  = BASE + "signature";
}
